### PR TITLE
feat: User list BF

### PR DESCRIPTION
**Pull request by nadiia-tsytsylina**
## Describe your changes

- updated account details page styles for admin and moved it to admin-panel;
- added login errors for inactive and deleted by admin users;
- added sorting, searching, deleting, updating, fetching users from db;
- added endpoint to get filtered users, updated user type;
- removed mocked data.

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://homecareaid.atlassian.net/browse/CC-234)

image
image

Снимок экрана 2024-01-11 160451
Снимок экрана 2024-01-11 162955
Снимок экрана 2024-01-11 163019

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

**Review by ProggerOleg — APPROVED** [review]

**Comment by Lilly-nn** [comment]
image
why is this error in the past?

**Comment by nadiia-tsytsylina** [comment]
> image why is this error in the past?

I asked Liuba about it. It was in acceptance criteria. But she didn't unswer yet

**Review by Lilly-nn — APPROVED** [review]

File: src/components/login-form/LoginForm.tsx [hunk 1]
```diff
@@ -54,8 +53,19 @@ function LoginForm(): JSX.Element {
...
+            setError('password', {
+              type: 'manual',
+              message: `${translate('loginForm.isDeletedByAdmin')}`,
+            });
+          } else if (errorMessage === translate('loginForm.isInactive')) {
+            setError('password', {
+              type: 'manual',
+              message: `${translate('loginForm.isInactive')}`,
+            });
+          } else if (error) {
```
Comment (Simpsonkk) [inline src/components/login-form/LoginForm.tsx:68]: why do we need these conditionals? You can just show an error message from server, you get different error messages from the server.

Comment (nadiia-tsytsylina) [reply src/components/login-form/LoginForm.tsx]: fixed

**Review by Simpsonkk — APPROVED** [review]

**Review by fedasevich — APPROVED** [review]

**Pull request merged**